lib/moab/storage_object_validator.rb

```diff
@@ -208,23 +222,22 @@ def error_code_msg(response_code, addl=nil)
 
     def check_required_manifest_files(dir, version)
       errors = []
-      has_manifest_inventory = File.exist?("#{dir}/#{MANIFEST_INVENTORY_PATH}")
-      has_signature_catalog = File.exist?("#{dir}/#{SIGNATURE_CATALOG_PATH}")
-      result = if has_manifest_inventory && has_signature_catalog
-                 nil
-               elsif has_manifest_inventory && !has_signature_catalog
-                 result_hash(NO_SIGNATURE_CATALOG, version)
-               elsif !has_manifest_inventory && has_signature_catalog
-                 result_hash(NO_MANIFEST_INVENTORY, version)
-               else
-                 result_hash(NO_FILES_IN_MANIFEST_DIR, version)
-               end
-      errors << result if result
+      unless contains_file?(File.join(dir, MANIFESTS_DIR))
```

Comment:
does File.join provide an efficiency boost or is it easier to read? vs concatenation?